keyboards/planck/keymaps/jweickm/keymap.c

```diff
@@ -0,0 +1,1079 @@
+                if (temp_mods & MOD_MASK_SHIFT) {
+                    register_code(KC_2);  // \"
+                } else {
+                    register_code16(KC_PIPE);  // /'
+                }
+            } else {
+                unregister_code(KC_2);
+                unregister_code16(KC_PIPE);
+            }
+            return true;
+        case DE_SCLN:
+            if (record->event.pressed) {
+                uint8_t temp_mods = get_mods() | get_oneshot_mods();
+                if (temp_mods & MOD_MASK_SHIFT) {
+                    tap_code(KC_DOT);  // ;
+                } else {
+                    add_mods(MOD_BIT(KC_LSFT));
+                    tap_code(KC_COMM);  // :
+                    del_mods(MOD_BIT(KC_LSFT));
+                }
+                return false;
+            }
+            break;
+        case DE_BSLS:
+            if (record->event.pressed) {
+                uint8_t temp_mods = get_mods() | get_oneshot_mods();
+                clear_mods();
+                clear_oneshot_mods();
+                add_mods(MOD_BIT(KC_RALT));
+                if (temp_mods & MOD_MASK_SHIFT) {
+                    register_code(KC_NUBS);  // | Pipe
+                } else {
+                    register_code(KC_MINS);  // Backslash
+                }
+                set_mods(temp_mods);
+            } else {
+                unregister_code(KC_NUBS);
+                unregister_code(KC_MINS);
+            }
+            return true;
+        // switch multiplexing for escape, short tap for escape, long press for context menu
+        case M_ESCM:
+            if (record->event.pressed) {
+                key_timer = timer_read();
+            } else {
+                if (timer_elapsed(key_timer) > TAPPING_TERM) {
+                    tap_code(KC_APP);
+                } else {
+                    tap_code(KC_ESC);
+                }
+            }
+            return false;
+        case M_RGUI_SCLN:
+            if (record->event.pressed) {
+                key_timer = timer_read();
+                add_mods(MOD_BIT(KC_RGUI));
+            } else {
+                del_mods(MOD_BIT(KC_RGUI));
+                if (timer_elapsed(key_timer) < (TAPPING_TERM * pinky_factor)) {
+                    uint8_t temp_mods = get_mods() | get_oneshot_mods();
+                    if (temp_mods & MOD_MASK_SHIFT) {
+                        tap_code(KC_DOT);  // ;
+                    } else {
+                        add_mods(MOD_BIT(KC_LSFT));
+                        tap_code(KC_COMM);  // :
+                        del_mods(MOD_BIT(KC_LSFT));
+                    }
+                } else {
+                    tap_code(KC_RGUI);
+                }
+            }
+            return false;
+            break;
+    }
+    return true;
+}
+
+bool     muse_mode      = false;
+uint8_t  last_muse_note = 0;
+uint16_t muse_counter   = 0;
+uint8_t  muse_offset    = 70;
+uint16_t muse_tempo     = 50;
+
+bool encoder_update(bool clockwise) {
```

Comment:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
```

Reply:
I didn't realize that this had changed. Thanks for the update!